SpatialGDK/Source/SpatialGDK/Public/Utils/ObjectAllocUtils.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+/**
+ * Helper to declare that a type cannot be allocated on the heap
+ * (it can still be stored in an array though)
+ * This should be enough to prevent types from becoming polymorphic.
+ * (or if they become polymorphic, it should only be in a context where
+ * the code responsible for storing it knows its final type, hence not needing a virtual dtor)
+ */
+class FNoHeapAllocation
```

Comment:
This name seems a little misleading because these types are used in a TArray which is heap allocated, maybe PlacementNewOnly or something? 

Is there a reason we need to explicitly disable heap allocating these types other than it's less than desirable to do so (new[] seems like it could be useful if TArray wasn't used)?

Reply:
PlacementNewOnly would convey the wrong information I think. As in "you are encouraged to do a placement new".
What I want to convey is that you are forbidden to use an allocator that would remember the alloced size from the pointer.

Reply:
Ah I see, that could be informative in the comment ^

Is this because these types are sometimes destroyed as another type?

Reply:
Thought about this some more and I think I get it, it would mean you could not transfer memory ownership to a smaller type (maybe except via slicing!?). Nice!